File: api/helpers/response.py
from flask import make_response, jsonify, Response
from typing import Union


# any changes in this file should be discussed
def success_response(data = None, errors: list = None, status: int = 200) -> Response:
    body = {
        "data": data,
        "errors": errors or [],
    }

    return make_response(
        jsonify(body),
        status
    )


def error_response(errors: Union[list, str] = None, status: int = 400) -> Response:
    if isinstance(errors, str):
        errors = [errors]

    body = {
        "data": None,
        "errors": errors or [],
    }

    return make_response(
        jsonify(body),
        status
    )

File: web/src/types/response.ts
import {CrossSearchResult, SearchResult} from "./index";

export type SuccessResponse<T> = {
  data: T | null;
  errors: string[] | null;
};

export type ErrorResponse<T> = {
  errors: string[];
  data: T | Record<string, any> | null;
};

export type SearchResponse = SuccessResponse<SearchResult[]>;
export type CrossSearchResponse = SuccessResponse<CrossSearchResult[]>;

